src/CoreWCF.Primitives/src/CoreWCF/Security/SecurityProtocolFactory.cs

```diff
@@ -0,0 +1,1085 @@
+                                {
+                                    expectKeyDerivation = true;
+                                }
+                            }
+                            SecurityTokenAttachmentMode mode = tokenAuthenticatorSpec.SecurityTokenAttachmentMode;
+                            if (mode == SecurityTokenAttachmentMode.SignedEncrypted
+                                || mode == SecurityTokenAttachmentMode.Signed
+                                || mode == SecurityTokenAttachmentMode.SignedEndorsing)
+                            {
+                                this.expectChannelSignedTokens = true;
+                                if (mode == SecurityTokenAttachmentMode.SignedEncrypted)
+                                {
+                                    this.expectChannelBasicTokens = true;
+                                }
+                            }
+                            if (mode == SecurityTokenAttachmentMode.Endorsing || mode == SecurityTokenAttachmentMode.SignedEndorsing)
+                            {
+                                this.expectChannelEndorsingTokens = true;
+                            }
+                        }
+                        this.channelSupportingTokenAuthenticatorSpecification =
+                            new ReadOnlyCollection<SupportingTokenAuthenticatorSpecification>((Collection<SupportingTokenAuthenticatorSpecification>)this.channelSupportingTokenAuthenticatorSpecification);
+                    }
+                }
+                VerifyTypeUniqueness(this.channelSupportingTokenAuthenticatorSpecification);
+                MergeSupportingTokenAuthenticators(timeoutHelper.RemainingTime());
+            }
+
+            //if (this.DetectReplays)
+            //{
+            //    if (!this.SupportsReplayDetection)
+            //    {
+            //        throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgument("DetectReplays", SR.GetString(SR.SecurityProtocolCannotDoReplayDetection, this));
+            //    }
+            //    if (this.MaxClockSkew == TimeSpan.MaxValue || this.ReplayWindow == TimeSpan.MaxValue)
+            //    {
+            //        throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.GetString(SR.NoncesCachedInfinitely)));
+            //    }
+
+            //    // If DetectReplays is true and nonceCache is null then use the default InMemoryNonceCache. 
+            //    if (this.nonceCache == null)
+            //    {
+            //        // The nonce needs to be cached for replayWindow + 2*clockSkew to eliminate replays
+            //        this.nonceCache = new InMemoryNonceCache(this.ReplayWindow + this.MaxClockSkew + this.MaxClockSkew, this.MaxCachedNonces);
+            //    }
+            //}
+
+            //this.derivedKeyTokenAuthenticator = new NonValidatingSecurityTokenAuthenticator<DerivedKeySecurityToken>();
```

Comment:
This is an important feature for IssuedToken support. Any reason why derivedTokenAuthenticator was removed?

Reply:
coming along with WindowsAuth.. I have already ported in my local... https://github.com/CoreWCF/CoreWCF/issues/259